pkg/stats/io_snapshots.go

```diff
@@ -37,25 +77,81 @@ func (h *snapshotHistory) SetInput(id string, latestInput json.Marshaler) {
 	h.latestInput[id] = latestInput
 }
 
-func (h *snapshotHistory) GetInput() ([]byte, error) {
+func (h *snapshotHistory) GetInputCopy() (map[string]interface{}, error) {
 	h.lock.RLock()
 	defer h.lock.RUnlock()
 	if h.latestInput == nil {
-		return []byte{}, nil
+		return map[string]interface{}{}, nil
 	}
-	result := map[string]interface{}{}
-	for id, obj := range h.latestInput {
-		jsn, err := obj.MarshalJSON()
-		if err != nil {
-			return nil, err
+	genericMaps, err := getGenericMaps(h.latestInput)
+	if err != nil {
+		return nil, err
+	}
+	return genericMaps, nil
+}
+
+// DEPRECIATED - Use GetFilteredInput or GetInputCopy instead
+func (h *snapshotHistory) GetInput() ([]byte, error) {
```

Comment:
Rather than deprecating, could we just have this wrap GetFilteredInput with no filters?

Reply:
Yes